### PR TITLE
feat(hogql): sparkline tags and title

### PR DESCRIPTION
**Pull request by mariusandra**
## Problem

Inspired by [this comment](https://github.com/PostHog/posthog/commit/7d7784251adbd3077cfbb66826de2d6e6cd95f7f#r143578812), why are sparklines (`select sparkline([1,2,3])`) not widely know?  I also find it kind of awkward that a simple function renders a chart. 

## Changes

While nothing changes with the `sparkline()` function, this PR adds a `<Sparkline />` component that echoes our lemon-ui sparkline, allowing for more customization:

image

This PR also makes objects work in HogQL, at least as something you pass through to the frontend.

## How did you test this code?

Added plenty of tests, played around in the UI.

## Next steps

Let's get this in and see if there's any love. We can easily revert or expand in the future.

### CODE REVIEW

**Review by daibhin — COMMENTED** [review]
One concern I have is that I might confuse the "React-like" components for our actual `Sparkline` React component. This is cool and I agree it's weird that the function generates a graph. The real problem for me here was writing the code that generated the data - a single function to make that easier would be a real win imo

**Comment by mariusandra** [comment]
> One concern I have is that I might confuse the "React-like" components for our actual `Sparkline` React component. This is cool and I agree it's weird that the function generates a graph. The real problem for me here was writing the code that generated the data - a single function to make that easier would be a real win imo

Let's take it all the way then:



image

image

File: mypy-baseline.txt [hunk 1]
```diff
@@ -209,7 +209,6 @@ posthog/hogql/resolver.py:0: error: Incompatible types in assignment (expression
 posthog/hogql/resolver.py:0: error: Argument 1 to "append" of "list" has incompatible type "BaseTableType | SelectUnionQueryType | SelectQueryType | SelectQueryAliasType | SelectViewType | None"; expected "SelectQueryType | SelectUnionQueryType"  [arg-type]
 posthog/hogql/resolver.py:0: error: Incompatible types in assignment (expression has type "Expr", variable has type "JoinExpr | None")  [assignment]
 posthog/hogql/resolver.py:0: error: Incompatible types in assignment (expression has type "Expr", variable has type "SampleExpr | None")  [assignment]
-posthog/hogql/resolver.py:0: error: Argument 2 to "convert_hogqlx_tag" has incompatible type "int | None"; expected "int"  [arg-type]
```
Comment (daibhin) [inline mypy-baseline.txt:212]: One down... some more to go

File: posthog/hogql/functions/sparkline.py [hunk 2]
```diff
@@ -4,9 +4,9 @@
 def sparkline(node: ast.Expr, args: list[ast.Expr]) -> ast.Expr:
     return ast.Tuple(
         exprs=[
-            ast.Constant(value="__hogql_chart_type"),
-            ast.Constant(value="sparkline"),
```
Comment (daibhin) [inline posthog/hogql/functions/sparkline.py:8]: Should we be getting rid of this existing method? I use it in error tracking (easily updated) but how can we know if there are other queries / insights relying on it

**Review by daibhin — COMMENTED** [review]
The HogQLX I'm not super familiar with but it all seems isolated and logical so I'm happy. Left one other question about deprecating the `sparkline` function

File: posthog/hogql/functions/sparkline.py (hunk 2 above)
Comment (mariusandra) [reply posthog/hogql/functions/sparkline.py:8]: we're not getting rid of it, it's just inlining the tag

Comment (daibhin) [reply posthog/hogql/functions/sparkline.py:8]: Ah yes, I re-read the description too

**Review by daibhin — APPROVED** [review]

File: mypy-baseline.txt (hunk 1 above)
Comment (mariusandra) [reply mypy-baseline.txt:212]: it seems about half of the errors here are from HogQL related files... so I'm going to have to take on this file some time when I'm back

**Pull request merged**